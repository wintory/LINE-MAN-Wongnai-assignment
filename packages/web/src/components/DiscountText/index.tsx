import { styled, Typography } from '@mui/material';
import { FC } from 'react';

interface DiscountTextProps {
  fullPrice: number;
  discountPrice: number;
}

const Price = styled('span')(({ theme }) => ({
  textDecoration: 'line-through',

  '&.discount': {
    pl: 2,
    textDecoration: 'none',
    color: theme.palette.error.main,
  },
}));

const DiscountText: FC<DiscountTextProps> = ({ fullPrice, discountPrice }) => {
  const price = discountPrice === 0 ? 'Free' : `${discountPrice} Baht`;

  return (
    <>
      <Typography variant="body1" color="common.black">
        price: <Price>{fullPrice || '-'}</Price>{' '}
        <Price className="discount">{price || '-'}</Price>
      </Typography>
    </>
  );
};

export default DiscountText;
