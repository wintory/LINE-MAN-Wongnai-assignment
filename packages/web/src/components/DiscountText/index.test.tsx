import { render } from '@testing-library/react';
import DiscountText from '.';

describe('DiscountText', () => {
  it('should render if it exists', () => {
    const { getByText } = render(
      <DiscountText fullPrice={100} discountPrice={60} />
    );

    expect(getByText('price:')).toBeTruthy();
    expect(getByText('100')).toBeTruthy();
    expect(getByText('60 Baht')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { container } = render(
      <DiscountText fullPrice={100} discountPrice={60} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
