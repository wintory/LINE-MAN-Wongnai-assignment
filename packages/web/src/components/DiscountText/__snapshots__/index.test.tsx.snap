// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountText should match snapshot 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-mjamk0-MuiTypography-root"
>
  price: 
  <span
    class="css-2660ae"
  >
    100
  </span>
   
  <span
    class="discount css-2660ae"
  >
    60 Baht
  </span>
</p>
`;
